pkg/apis/pipeline/v1beta1/sanity_test.go

```diff
@@ -23,7 +23,7 @@ import (
 	"reflect"
 	"testing"
 
-	"github.com/knative/build-pipeline/test"
+	"github.com/knative/build-pipeline/testlib"
```

Comment:
> moves test/* -> testlib/* as knative/* repos follow a pattern where prow executes a script in test/presubmit-test.sh

I would rather have both - we can keep the Go files with the bash scripts in one dir.

I'd rather avoid a name like `testlib` cuz it reminds me of [the go recommendations about package names](https://blog.golang.org/package-names#TOC_5.), to avoid names like `lib` and `util`.